### PR TITLE
Tidying OMERO developer docs

### PR DESCRIPTION
**Pull request by hflynn**
Various bits and bobs that came up as I was reviewing our developer docs to catalogue our examples.

### CODE REVIEW

**Comment by hflynn** [comment]
Also see https://trello.com/c/MNIlsSUs/290-5-1-extension-points

**Comment by hflynn** [comment]
https://www.openmicroscopy.org/site/support/omero5.1/developers/Server/CollectionCounts.html looks ancient (referencing 3.0Beta3!!) Is there anything worth saving here or shall I just delete the whole page?

**Comment by hflynn** [comment]
Also removed a large chunk of stuff about backwards compatibility to pre 3.0beta

**Comment by mtbc** [comment]
The collection counts explanation / example may be worth putting somewhere among developer docs but yes, that page does generally seem to need some TLC / obsoleting.

File: omero/developers/Server/Sessions.txt [hunk 1]
```diff
@@ -1,25 +1,8 @@
...
-   specified.
-#. Passwords were sent in the clear on every invocation.
-#. Sql queries on every invocation caused significant overhead.
-
-Blitz did not suffer from these problems, but the login functionality
-was largely outside of the core server code and sessions were more
-volatile: a loss of an Ice connection caused all resources to be lost.
-With |OmeroSessions|, both login systems have been brought together and 
-simplified.
+OMERO sessions simplifies the handling of login session for
```
Comment (rleigh-dundee) [inline omero/developers/Server/Sessions.txt]: "a login session" or "login sessions"?

**Comment by rleigh-dundee** [comment]
Looks generally fine and good to merge.

**Comment by hflynn** [comment]
Thanks, I'll just push a fix for the login session typo.

**Comment by hflynn** [comment]
@joshmoore do you want more work here re: extension points or is this good to merge from you too?

**Comment by joshmoore** [comment]
I'd go with it. There might be something to be said for eventually make the whole workflow a bit easier, but that may come naturally as a part of decoupling (i.e. when omero-api and omero-model are separate repos)

**Pull request merged**